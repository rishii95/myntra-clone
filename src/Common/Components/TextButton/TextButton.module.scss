.textButton {
    border: none;
    background-color: inherit;
    cursor: pointer;
}