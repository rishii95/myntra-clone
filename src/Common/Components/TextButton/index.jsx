import React from 'react';
import { PropTypes } from 'prop-types';
import { Button } from 'antd';
import variables from '../../variables.module.scss';

export default function TextButton(props) {
  const {
    children, weight, color, size,
  } = props;
  const getStyles = () => ({ color, fontWeight: weight, fontSize: size });
  return (
    <Button data-testid="text-button" type="link" style={getStyles()}>{children}</Button>
  );
}

TextButton.propTypes = {
  children: PropTypes.node.isRequired,
  weight: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.string,
};

TextButton.defaultProps = {
  weight: variables.mediumFont,
  color: variables.pink,
  size: variables.subtitleSize,
};
