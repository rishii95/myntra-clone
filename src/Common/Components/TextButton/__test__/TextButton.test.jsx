/* eslint-disable no-undef */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TextButton from '../index';

afterEach(cleanup);
test('it matches snapshot', () => {
  const { container } = render(<TextButton>Hello</TextButton>);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders button correctly', () => {
  const { getByTestId } = render(<TextButton>Click me</TextButton>);
  expect(getByTestId('text-button')).toHaveTextContent('Click me');
});

test('renders button correctly', () => {
  const { getByTestId } = render(<TextButton>Press</TextButton>);
  expect(getByTestId('text-button')).toHaveTextContent('Press');
});
