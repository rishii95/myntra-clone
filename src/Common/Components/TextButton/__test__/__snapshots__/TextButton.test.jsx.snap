// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<button
  class="ant-btn ant-btn-link"
  data-testid="text-button"
  style="color: pink; font-weight: mediumFont;"
  type="button"
>
  <span>
    Hello
  </span>
</button>
`;
